tests/model_formsets/test_uuid.py

```diff
@@ -30,3 +38,135 @@ def test_inlineformset_factory_ignores_default_pks_on_submit(self):
             'uuidpkchild_set-2-name': '',
         })
         self.assertTrue(formset.is_valid())
+
+    def test_inlineformset_factory_nulls_default_pks_uuid_parent_auto_child(self):
+        """
+        #24958 - If we're adding a new object, a parent's auto-generated pk
+        from the model field default should be ignored as it's regenerated on
+        the save request.
+
+        Tests the case of a parent object with a UUID primary key and a child
+        object with an AutoField primary key.
+        """
+        FormSet = inlineformset_factory(UUIDPKParent, AutoPKChildOfUUIDPKParent,
+                                        fields='__all__')
+        formset = FormSet()
+        self.assertIsNone(formset.forms[0].fields['parent'].initial)
+
+    def test_inlineformset_factory_ignores_default_pks_on_submit_uuid_parent_auto_child(self):
```

Comment:
The "nulls defaults" test is for the statement under `If we're adding the related instance, ignore its primary key` in `forms/models.py` as seen in 1306cd1e8acfb13602ee8dc40993b2505cd7523b. We probably don't need the additional tests for that.